src/networkd.c

```diff
@@ -1202,6 +1202,25 @@ write_wpa_conf(const NetplanNetDefinition* def, const char* rootdir, GError** er
                     break;
                 case NETPLAN_WIFI_MODE_ADHOC:
                     g_string_append(s, "  mode=1\n");
+
+                    if (ap->channel) {
```

Comment:
This looks fine.

Does it make sense to still have the property `freq_list` when operating in ad-hoc mode? The documentation says `freq_list` are `frequencies in MHz to allow for selecting the BSS`. As an interface in ad hoc mode will not connect to BSSes, this property will probably be ignore (but I'm not sure). If that is the case, maybe we should not emit `freq_list` for an ad hoc interface. What do you think?

Reply:
aggree, i should not emit `freq_list` in adhoc mode. 👍
it was ok for me as a quick fix, because having `freq_list` does not break thngs in adhoc mode. i will take care of it in the next days.